### PR TITLE
[vtadmin] cell apis

### PR DESCRIPTION
**Pull request by ajm188**
## Description

vtadmin variants of:
- `GetCellInfoNames` (via `/cells?names_only=true`)
- `GetCellInfo`
- `GetCellsAliases`

### Demo

```
➜  local git:(andrew/vtadmin/cell-apis) ✗ curl -s localhost:14200/api/cells\?cluster\=local | jq '.'
{
  "result": {
    "cell_infos": [
      {
        "cluster": {
          "id": "local",
          "name": "local"
        },
        "name": "zone1",
        "cell_info": {
          "server_address": "localhost:2379",
          "root": "/vitess/zone1"
        }
      }
    ]
  },
  "ok": true
}
```

## Related Issue(s)

#8705

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required -- TODO!
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtadmin/cluster/cluster.go [hunk 1]
```diff
@@ -768,6 +768,116 @@ func (c *Cluster) getShardSets(ctx context.Context, keyspaces []string, keyspace
...
+// GetCellInfos returns a list of ClusterCellInfo objects for cells in the
+// given cluster.
+//
+// If req.Cells is set, cells are restricted only to cells with those names.
+// Note: specifying a cell name that does not exist in the cluster fails the
+// overall request.
+//
+// If req.NamesOnly is set, each ClusterCellInfo will only contain the Cluster
+// and Name fields. req.Cells takes precedence over this option.
+func (c *Cluster) GetCellInfos(ctx context.Context, req *vtadminpb.GetCellInfosRequest) ([]*vtadminpb.ClusterCellInfo, error) {
```
Comment (doeg) [inline go/vt/vtadmin/cluster/cluster.go:780]: Obligatory question of "do we need tests for either of these"

File: go/vt/vtadmin/cluster/cluster.go [hunk 2]
```diff
@@ -768,6 +768,116 @@ func (c *Cluster) getShardSets(ctx context.Context, keyspaces []string, keyspace
...
+		resp, err := c.Vtctld.GetCellInfoNames(ctx, &vtctldatapb.GetCellInfoNamesRequest{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to GetCellInfoNames: %w", err)
+		}
+
+		names = resp.Names
+	}
+
+	namesOnly := req.NamesOnly
+	if namesOnly {
```
Comment (doeg) [inline go/vt/vtadmin/cluster/cluster.go:799]: Just a question about """idiomatic go""": isn't this the same as `if namesOnly && len(req.Cells) != 0`?

**Review by doeg — COMMENTED** [review]
Thank you for doing this!!! Obligatory blocking question about tests but otherwise this is :chef-kiss: and I super appreciate it.

File: go/vt/vtadmin/cluster/cluster.go (hunk 2 above)
Comment (ajm188) [reply go/vt/vtadmin/cluster/cluster.go:799]: yuppppp (i originally had An Thought about doing the early return namesOnly loop in this section too, but never came back here), thanks!

File: go/vt/vtadmin/cluster/cluster.go (hunk 1 above)
Comment (ajm188) [reply go/vt/vtadmin/cluster/cluster.go:780]: yes and yes

Comment (ajm188) [reply go/vt/vtadmin/cluster/cluster.go:780]: annnd done

**Review by doeg — APPROVED** [review]

**Comment by ajm188** [comment]
i'm gonna wait for #10233 and rebase, because i added a `Dial` call here that'll need to get cleaned up, if that's okay with you!

**Pull request merged**